### PR TITLE
Xenobio tweaks and Pet carriers!

### PR DESCRIPTION
**Pull request by FartMaster69420**
So, this adds a few buffs for xenobiology along with a new pet carrier item (Ported from TGcode)

Source of the TG PR: https://github.com/tgstation/tgstation/pull/33231

For xenobiology:
-Research biosuit lockers will now actually spawn with the xenobiology bag.

-Xeno Tasers can now be used to stun simplemobs, the amount of hits required to stun is calculated off of the creature's current health divided by 15. Stuns will apply the weakened effect to the mob for a short duration, but deal no damage. The hits must be made fairly quickly one after the other or the counter for hits made will reset.

-The monkey gun and portable slime grinder had their recipes altered to remove the need for morphium.

-Wildlife Stasis Cages were added to cargo for 25 points, this will allow xenobio and exploration to capture creatures on maps without mapped in stasis crates.

-A bluespace stasis crate was added, constructable from science with biotech level 8. It is a subtype of the pet carriers, and allows easier transport of dangerous mobs. This item can be placed into a backpack, so to counter this obvious advantage it uses morphium in it's construction and requires higher than average biotech.

Pet Carriers!

-Constructable from any autolathe!

-Can hold up to 3 tiny mobs (EG: mice), or 1 small mob (Catslug, monkey, doggo, ect)

-Can be locked to secure the critter inside and prevent escape.

-Locked critters can escape via resist to unlock the crate, and another resist to leave.

-The OOC Escape verb can also be used to instantly teleport out of a crate if you don't want to be there.

-People can be put in pet carriers! You can scoop anyone 20% smaller than yourself and put them in the pet carrier. Standard pickup prefs are respected for this interaction.

### CODE REVIEW

File: code/game/objects/items/weapons/pet_carrier.dm [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	var/locked = FALSE
+	var/list/occupants = list()
+	var/occupant_weight = 0
+	var/max_occupants = 3 //Hard-cap so you can't have infinite mice or something in one carrier
+	var/max_occupant_weight = MOB_SMALL //This is calculated from the mob sizes of occupants
+
+/obj/item/weapon/pet_carrier/New()
+	verbs += .proc/drop_pet
+	update_icon()
+	..()
```
Comment (silicons) [inline code/game/objects/items/weapons/pet_carrier.dm:33]: this really should use Initialize(mapload) insead of new

```
/obj/item/weapon/pet_carrier/Initialize(mapload)
    verbs += /obj/item/weapon/pet_carrier/proc/drop_pet
    update_icon()
    return ..()
```

File: code/game/objects/items/weapons/pet_carrier.dm [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+/obj/item/pet_carrier/MouseDrop(atom/over_atom)
+	. = ..()
+	var/turf/T = get_turf(over_atom.loc)
+	if(T.check_density(ignore_mobs = TRUE) && usr.CanUseTopic(src, TRUE) && usr.Adjacent(over_atom) && open && occupants.len)
+		usr.visible_message(span_notice("[usr] unloads [src]."), \
+		span_notice("You unload [src] onto [over_atom]."))
+		for(var/V in occupants)
+			remove_occupant(V, T)
+*/
+/obj/item/weapon/pet_carrier/proc/drop_pet()
```
Comment (silicons) [inline code/game/objects/items/weapons/pet_carrier.dm:198]: this doesn't need to be a /proc/ that's added at initialize unless you intend to remove it later
verbs are intrinsically just what you did here (verbs are the exact same as a proc in byond!) only it basically adds itself automatically (yet can also be removed later)

File: code/game/objects/items/weapons/pet_carrier.dm [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+/obj/item/weapon/pet_carrier/proc/remove_occupant(mob/living/occupant, turf/new_turf)
+	if(!(occupant in occupants) || !istype(occupant))
+		return
+	occupant.forceMove(new_turf ? new_turf : drop_location())
+	if(occupant.buckled && istype(occupant.buckled, /obj/effect/energy_net))
+		occupant.buckled.dropInto(src)
+	occupants -= occupant
+	occupant_weight -= occupant.mob_size
+	occupant.dir = SOUTH
```
Comment (silicons) [inline code/game/objects/items/weapons/pet_carrier.dm:240]: use setDir so relevant COMSIG's can fire (assuming upstream implemented that)

Comment (FartMaster69420) [reply code/game/objects/items/weapons/pet_carrier.dm:240]: setDir doesn't exist in this codebase, unless it's called something goofy and I can't find it.

**Comment by Very-Soft** [comment]
I'm cool with some parts of this and not cool with others. I need to test and see how some of these things feel and I don't have time right now!

Basically, simplemobs are allllready pretty easy to cheese if you know what you are doing, and this seems like just more tools to do more of that. I am not so sure I want there to be more ways to permastun mobs so you can instantly discard stuff into a magic box that they can't escape from. 

Pet carriers sound cute, and their player interaction seems fine, but seem like they would be more likely used to cheese stuff, and I'm not so sure I want people running up to the angry jellyfish or whatever and just scooping it up inside a funny box.

**Comment by FartMaster69420** [comment]
> I'm cool with some parts of this and not cool with others. I need to test and see how some of these things feel and I don't have time right now!
> 
> Basically, simplemobs are allllready pretty easy to cheese if you know what you are doing, and this seems like just more tools to do more of that. I am not so sure I want there to be more ways to permastun mobs so you can instantly discard stuff into a magic box that they can't escape from.
> 
> Pet carriers sound cute, and their player interaction seems fine, but seem like they would be more likely used to cheese stuff, and I'm not so sure I want people running up to the angry jellyfish or whatever and just scooping it up inside a funny box.

At the moment hostile mobs can only be placed into the carrier (and even then, only the 'advanced science' version) if they're caught in a net. Even stunning with the xenotaser isn't enough to crate them.

And only the bluespace carrier will put mobs into stasis to prevent attacks. Mobs placed in normal carriers can continue to attack people near the crate. Even though that's an edge case, since most hostile mobs are too big for the normal carrier anyway.

As far as xenotaser stuns I'm cool with reworking it a bit. (Maybe make it have a chance of making a mob flee instead of stunning?)

File: code/game/objects/items/weapons/pet_carrier.dm (hunk 3 above)
Comment (Zeldazackman) [reply code/game/objects/items/weapons/pet_carrier.dm:239]: Necro post, but setDir is set_dir on here

**Comment by Cameron653** [comment]
Doing PR cleanup and this has been sitting for 4 months. 

Feel free to re-open the PR with conflicts fixed if more discussion is to be had.

**Pull request closed without merging**